tests/test_websockets.py

```diff
@@ -315,6 +315,30 @@ async def asgi(receive, send):
         assert websocket.extra_headers == [(b"additional", b"header")]
 
 
+def test_no_additional_headers(test_client_factory):
+    expected_headers = []
+
+    def wrap(send):
+        async def _send(message):
+            if "headers" in message:
+                actual_headers = message.get("headers")
+                assert actual_headers == expected_headers
+            return await send(message)
+        return _send
+
+    def app(scope):
+        async def asgi(receive, send):
+            websocket = WebSocket(scope, receive=receive, send=wrap(send))
+            await websocket.accept(headers=None)
+            await websocket.close()
+
+        return asgi
+
+    client = test_client_factory(app)
+    with client.websocket_connect("/"):
+        pass  # pragma: nocover
```

Comment:
```suggestion
def test_no_additional_headers(test_client_factory):
    def app(scope):
        async def asgi(receive, send):
            websocket = WebSocket(scope, receive=receive, send=send)
            await websocket.accept(headers=None)
            await websocket.close()

        return asgi

    client = test_client_factory(app)
    with client.websocket_connect("/") as websocket:
        assert websocket.extra_headers == []
```

Reply:
I think this will do that.

Reply:
Yes, I agree that it's better to create a dedicated test case. Thanks for the suggestion, it's much more clear.